sdk/src/handshake.rs

```diff
@@ -13,16 +13,32 @@ use crate::CapEnv;
 /// creates the root bucket for this contract and gives it `creation_cycles`
 /// cycles.
 pub fn handshake(creation_cycles: u64, router_override: Option<Principal>) {
+    let mut controllers = Vec::new();
+
     let arg = management::CreateCanisterArgument { settings: None };
 
     let router = {
         if let Some(router_override) = router_override {
+            controllers.push(router_override);
             Router::new(router_override)
         } else {
-            Router::new(Principal::from_str("lj532-6iaaa-aaaah-qcc7a-cai").unwrap())
+            let p = Principal::from_str("lj532-6iaaa-aaaah-qcc7a-cai").unwrap();
+            controllers.push(p);
+            Router::new(p)
         }
     };
 
+    let create_settings = management::CanisterSettings {
+        controllers: Some(controllers),
+        compute_allocation: None,
+        memory_allocation: None,
+        freezing_threshold: None,
+    };
```

Comment:
```rs
    let router_pid = router_override.unwrap_or_else(|| Principal::from_str("lj532-6iaaa-aaaah-qcc7a-cai").unwrap());
    let router = Router::new(router_pid);

    let create_settings = management::CanisterSettings {
        controllers: Some(vec![router_pid]),
        compute_allocation: None,
        memory_allocation: None,
        freezing_threshold: None,
    };
```

Reply:
Sweet thanks :+1: . Fixed.  This is good to merge now.